iree/turbine/kernel/wave/templates/paged_decode_attention.py

```diff
@@ -88,20 +91,19 @@ def phase_0_constraints():
         )
         wg_func = lambda wg: wg * sympy.floor(T / U) + sympy.Min(wg, sympy.Mod(T, U))
         constraints += [
-            tkw.WorkgroupConstraint(
-                T, BLOCK_T, 0, apply_fn=lambda wg: wg_func(wg), primary=False
-            )
+            tkw.WorkgroupConstraint(T, BLOCK_T, 0, apply_fn=wg_func, primary=False)
         ]
         constraints += [tkw.TilingConstraint(T, 1, iters=count)]
 
         # BH is the kv-head index and is distributed across workgroups.
         # B is the query index and is distributed like BH but with a different
         # workgroup and wave tile size.
-        # TODO: We will want to add a function to the workgroup constraint to
-        # allow for using WG / ceil(kv_group_num, BLOCK_B) instead of just WG.
-        # This can be done by adding an optional additional argument to the WorkgroupConstraint.
 
-        constraints += [tkw.WorkgroupConstraint(BH, BLOCK_BH, 1)]
+        wg_func_2 = lambda wg: wg // math.ceil(head_ratio / HEAD_BLOCK_SIZE)
```

Comment:
Can we get a little bit of doc on what is going on here? :)

Reply:
sure, but short summary is that this is GQA.

Reply:
make sense! :)

Reply:
added comment